### ISSUE TITLE
Don't submit answers to skipped questions

### ISSUE THREAD

**Issue by ttavenner** [Code4HR/okcandidate#72]
If someone selects a value or an intensity that value is automatically added to the response array, even if they click the skip button. This can result in a database error if someone clicks an intensity but no answer.

**Comment by stanzheng** [comment]
i did something similar here where you would be able to null out the answer but the dispatcher doesn't fire, maybe i went down the wrong path on this.

```
                if (action.type === SELECT_SURVEY_QUESTION_RESPONSE_INTENSITY) {
                  response.intensity == action.intensity ? response.intensity = undefined : response.intensity = action.intensity
                }
                else if (action.type === SELECT_SURVEY_QUESTION_RESPONSE) {  
                 response.answerId == action.intensity ? response.answerId = undefined: response.answerId = action.answer.id
                }
```

**Comment by wbprice** [comment]
The fast way to solve this problem is to run a filter on responses before they are posted to the backend.  This could be done in the action generator that makes the API call at https://github.com/Code4HR/okcandidate/blob/master/assets/js/redux/survey/survey-actions.js#L150

Psedocode looks like:
```
responses.filter(response => {
  return response.answerId && response.intensity
})
```

**Comment by kmcurry** [comment]
Except that you have to "Skip" every answer. I assume under the hood you
mean something like "unanswered and not rated"?

On Thursday, April 7, 2016, W. Blaine Price <notifications@github.com>
wrote:

> The fast way to solve this problem is to run a filter on responses before
> they are posted to the backend. This could be done in the action generator
> that makes the API call at
> https://github.com/Code4HR/okcandidate/blob/master/assets/js/redux/survey/survey-actions.js#L150
>
> Psedocode looks like:
>
> responses.filter(response => {
>   return response.answerId && response.intensity
> })
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Code4HR/okcandidate/issues/72#issuecomment-206916630>
>


-- 
Kevin Curry
Virginia Beach, VA
https://comparityins.com

**Comment by ttavenner** [comment]
The original issue is that right now you can select an answer and/or an intensity and then click the "Skip" button. That answer will still be submitted to the survey. You can also partially answer a question by either selecting an answer with no intensity or an intensity with no answer.

**Comment by stanzheng** [comment]
resolved by #82

**Comment by ryayak1460** [comment]
I should probably reference the issue in these pull requests, eh?

On 7 April 2016 22:36:06 GMT-04:00, Stanley Zheng <notifications@github.com> wrote:
>resolved by #82 
>
>---
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/Code4HR/okcandidate/issues/72#issuecomment-207181435

-- 
Ryan Y.
Software Design Engineer
Alion Science and Technology Corporation
GnuPG Key: 0x358CD775
VoIP (ZRTP-encrypted): ryayak1460@ostel.co
IM (OTR encryption): ryayak1460@dukgo.com
